docs/wiki/Vision.md

```diff
@@ -0,0 +1,115 @@
+- The deployed **environments** that can be individual services, compositions such as workloads, or entire landing zones and the like
+- The **orchestration** deploying the modules in the form of template/pipeline orchestration using for example GitHub actions or Azure DevOps pipelines
+  - No matter the platform we can differentiate two different deployment approaches:
+    - **_Template Orchestration_**: These types of deployments reference individual modules from a 'main/environment' bicep/ARM template and use its capabilities to pass parameters & orchestrate the deployments. By default, deployments are deployed in parallel by the Azure Resource Manager while accounting for all dependencies defined. Furthermore, the deploying pipeline only needs one deployment job that triggers the template's deployment.
+    - **_Pipeline Orchestration_**: This approach uses the platform specific pipeline capabilities (for example pipeline jobs) to trigger the deployment of individual modules, where each job deploys one module. By defining dependencies in between jobs you can make sure your resources are deployed in order. Parallelization is achieved by using a pool of pipeline agents that execute the jobs, while accounting for all dependencies defined.
+- The fundamental **modules** that each deploy a service, i.e. are the building blocks to deploy environments
+
+To make each component a bit more tangible, let's take a look at the following example:
+
+- Target Environment
+  - A virtual machine connected to a storage account
+- Orchestration
+  - Template-orchestration using GitHub actions
+- Modules
+  - Resource Group
+  - Virtual Network
+  - Virtual Machine
+  - Storage Account
+
+For this example we could create a GitHub workflow that signs into Azure and run's a single deployment using for example the PowerShell command `New-AzResourceGroupDeployment`.
+Furthermore we'd then create an orchestration-template the deploys the above resources in the following parallel groups (using dependencies)
+1. The resource group
+1. The virtual network & storage account
+1. The virtual Machine
+
+Then we'd only need to create a parameter file for the orchestration-template and have the workflow deploy both in combination.
+
+### What is the intended deployment flow?
+
+In this section we'll take a deeper look into the fundamental flow of the platform.
+
+First things first, we would work towards the deployment of our environments in 3 phases:
+
+<img src="media/deploymentFlow.png" alt="Deployment flow" height="150">
+
+The **Produce components** phase is the phase where you're supposed to set up your modules and validate that they work. In case of _CARML_ this means we have a pipeline for each module that, with one or multiple test-parameter files, runs static as well as validation & deployment tests on its template. Once if all tests pass we can be certain no bugs were introduced (in case the template was altered) and all features work as intended.
```

Comment:
Consider rephrasing to make it clearer

Reply:
done

Reply:
just one typo: 'prove' their correctness.

Reply:
fixed